src/services/utilServices/MailClient.ts

```diff
@@ -28,16 +35,57 @@ class MailClient {
     }
 
     try {
-      await axios.post(endpoint, email, {
+      const sendMailResponse = await axios.post<MailData>(sendEndpoint, email, {
         headers: {
           Authorization: `Bearer ${this.POSTMAN_API_KEY}`,
         },
       })
+      this.verifyMail(sendMailResponse.data)
     } catch (err) {
       logger.error(`Error occurred when sending email to ${recipient}: ${err}`)
       throw new Error("Failed to send email.")
     }
   }
+
+  async verifyMail(sendMailData: MailData): Promise<void> {
+    const verfyEndpoint = `${POSTMAN_API_URL}/transactional/email/${sendMailData.id}`
+    await new Promise((res) => setTimeout(res, MAIL_VERIFICATION_DELAY))
+    const verifyMailResponse = await axios.get<MailData>(verfyEndpoint, {
+      headers: {
+        Authorization: `Bearer ${this.POSTMAN_API_KEY}`,
+      },
+    })
+    this.logEmailStatus(verifyMailResponse.data)
+  }
+
+  logEmailStatus(verifyMailData: MailData) {
+    const mailStatus = verifyMailData.status
+    switch (mailStatus) {
+      case "UNSENT":
+        logger.error(
+          `Email to ${verifyMailData.recipient} not accepted: ${verifyMailData}`
+        )
+        break
+      case "ACCEPTED":
+        logger.error(
+          `Email to ${verifyMailData.recipient} not sent: ${verifyMailData}`
+        )
+        break
+      case "SENT":
+        logger.error(
+          `Email to ${verifyMailData.recipient} not delivered: ${verifyMailData}`
+        )
+        break
+      case "BOUNCED":
+        logger.error(
+          `Email to ${verifyMailData.recipient} rejected by recipient's mail server: ${verifyMailData}`
+        )
+        break
+      default:
+        logger.info(`Email delivered to ${verifyMailData.recipient}`)
+        break
```

Comment:
we can use a `never` variable if we go the sum type route: 
```ts
const err: never = mailStatus
throw new err(...)
```

this will get caught at compile time as `never` cannot be assigned to a concrete type - we're making use of a trick here to get it

Reply:
okay, listing all valid status in switch and changing default to unknown status.